From f43b79b3f5034ddda1da1855e744eb02079b73b8 Mon Sep 17 00:00:00 2001
From: Adriana Reus <adriana.reus@nxp.com>
Date: Tue, 6 Mar 2018 15:28:03 +0200
Subject: [PATCH 3433/5242] MLK-17696: Change resource from PID_0 to SYSTEM
 for scu control.

commit  543caec053ee884f78cfc27ba26bb3c23725c954 from
https://source.codeaurora.org/external/imx/linux-imx.git

Syncs with the following change in scfw.

    Author: Chuck Cannon <chuck.cannon@freescale.com>
    Date:   Mon Mar 5 07:44:27 2018 -0600

    SCF-22: Move SCU controls to SYSTEM. Allows AP to use SCU temp
    sensor.

BuildInfo:
  - SCFW e56d4c0a, IMX-MKIMAGE d4e440b2, ATF 4af5ca0
  - U-Boot 2017.03-imx_v2017.03_4.9.51_imx8_beta2+gf195c38

Tested-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Nitin Garg <nitin.garg@nxp.com>
Signed-off-by: Adriana Reus <adriana.reus@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/soc-imx8.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/imx/soc-imx8.c b/drivers/soc/imx/soc-imx8.c
index d5d922c..941945e 100644
--- a/drivers/soc/imx/soc-imx8.c
+++ b/drivers/soc/imx/soc-imx8.c
@@ -118,7 +118,7 @@ static u32 imx_init_revision_from_scu(void)
 		return IMX_CHIP_REVISION_UNKNOWN;
 	};
 
-	sc_err = sc_misc_get_control(ipc_handle, SC_R_SC_PID0, SC_C_ID, &id);
+	sc_err = sc_misc_get_control(ipc_handle, SC_R_SYSTEM, SC_C_ID, &id);
 	if (sc_err != SC_ERR_NONE) {
 		WARN(1, "%s: Cannot get control\n", __func__);
 
-- 
1.7.9.5

